### ISSUE TITLE
Problem with ZONEMASTER_RECORD

### ISSUE THREAD

**Issue by mattias-p** [zonemaster/zonemaster-engine#466]
When I run `ZONEMASTER_RECORD=1 prove -Ilib` I get a whole lot of these:
```
...
Use of uninitialized value in subroutine entry at /home/mattiasp/zonemaster-engine/lib/Zonemaster/Engine/Nameserver.pm line 313, <DATA> line 1.
Use of uninitialized value in subroutine entry at /home/mattiasp/zonemaster-engine/lib/Zonemaster/Engine/Nameserver.pm line 382, <DATA> line 1.
...
```

Edit: Maybe I'm using an old version of Zonemaster::LDNS.

**Comment by vlevigneron** [comment]
I do not see that. Here is my results (just the beginning, perhaps I should run the whole test suite...). There are some issues but it is just because that some 'intentionally broken zone' are no longer 'broken'... :-)

```
> prove -Ilib
t/00-load.t ............... 1/? # Testing Zonemaster Engine v2.0.12, Perl 5.022004, /home/levigner/perl5/perlbrew/perls/perl-5.22.4/bin/perl
t/00-load.t ............... ok   
t/Test-address.t .......... ok    
t/Test-basic.t ............ 19/? 
#   Failed test 'HAS_NAMESERVERS'
#   at t/Test-basic.t line 97.

#   Failed test 'HAS_NAMESERVER_NO_WWW_A_TEST'
#   at t/Test-basic.t line 98.
t/Test-basic.t ............ 24/? # Looks like you failed 2 tests of 34.
t/Test-basic.t ............ Dubious, test returned 2 (wstat 512, 0x200)
Failed 2/34 subtests 
	(less 5 skipped subtests: 27 okay)
t/Test-connectivity.t ..... 18/? 
#   Failed test 'Nameservers with Uniq AS'
#   at t/Test-connectivity.t line 55.
t/Test-connectivity.t ..... 20/? 
#   Failed test 'Nameserver UDP port 53 unreachable'
#   at t/Test-connectivity.t line 59.

#   Failed test 'Nameserver TCP port 53 unreachable'
#   at t/Test-connectivity.t line 60.
t/Test-connectivity.t ..... 24/? # Looks like you failed 3 tests of 29.
t/Test-connectivity.t ..... Dubious, test returned 3 (wstat 768, 0x300)
Failed 3/29 subtests 
t/Test-consistency.t ...... ok    
```

**Comment by matsduf** [comment]
Fixed by #473.
